ribbon/src/main/java/feign/ribbon/RibbonClient.java

```diff
@@ -1,31 +1,40 @@
 package feign.ribbon;
 
-import com.netflix.client.ClientException;
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
 import com.netflix.client.ClientFactory;
 import com.netflix.client.config.CommonClientConfigKey;
 import com.netflix.client.config.DefaultClientConfigImpl;
 import com.netflix.client.config.IClientConfig;
 import com.netflix.loadbalancer.ILoadBalancer;
 
-import java.io.IOException;
-import java.net.URI;
-
 import feign.Client;
 import feign.Request;
 import feign.Response;
 
 /**
- * RibbonClient can be used in Fiegn builder to activate smart routing and resiliency capabilities
+ * RibbonClient can be used in Feign builder to activate smart routing and resiliency capabilities
  * provided by Ribbon. Ex.
+ * <p>
+ * 
  * <pre>
- * MyService api = Feign.builder.client(new RibbonClient()).target(MyService.class,
- * "http://myAppProd");
+ * MyService api = Feign.builder.client(RibbonClient.DEFAULT).target(MyService.class,
+ *     &quot;http://myAppProd&quot;);
  * </pre>
+ * 
+ * <p>
  * Where {@code myAppProd} is the ribbon client name and {@code myAppProd.ribbon.listOfServers}
  * configuration is set.
  */
 public class RibbonClient implements Client {
 
+  public static RibbonClient DEFAULT = new RibbonClient();
+  
+  private volatile Map<String, LBClient> lbClientCache = new HashMap<String, LBClient>();
```

Comment:
nit: always linkedhashmap
Discussed in the past, but this has consistent ordering which costs almost nothing. when tests are written and/or JREs switch, this pays dividends.